### PR TITLE
Support channel selection for multichannel CZI files

### PR DESCRIPTION
**Pull request by hrehmaan**
If the CZI file contains multiple channels and the user does not specify which channel to use, the loader raises a clear error message:

ValueError: This CZI file contains 2 channels. Available channels are 0 to 1. Please select one channel using the --channel option, for example: mps2mp4 Control_1.czi --channel 1

The user can then select any available channel explicitly, for example:
```bash
mps2mp4 Control_1.czi --channel 0
```
For single-channel CZI files, no channel argument is required, and the file is loaded normally without raising an error.

Issue #94

### CODE REVIEW

File: src/mps/load.py [hunk 1]
```diff
@@ -258,11 +361,115 @@ def load_czi(fname):
...
+            "Expected 3D data after squeezing or 4D data with channels."
+        )
+
+    # Move time axis to the last axis
+    if time_axis != images.ndim - 1:
+        images = np.moveaxis(images, time_axis, -1)
+
+    # Now images should be spatial_x/spatial_y/time, but CZI often gives y/x/time.
+    # Match the MPS convention: frames.shape == (x, y, time)
+    frames = np.swapaxes(images, 0, 1)
```
Comment (finsberg) [inline src/mps/load.py]: This seems to be repeating the code in `_format_czi_frames`. Should we use this function instead?

File: src/mps/load.py [hunk 2]
```diff
@@ -484,9 +691,9 @@ def load_tiff(fname):
     )
 
 
-def load_file(fname, ext):
+def load_file(fname, ext, channel=None):
```
Comment (finsberg) [inline src/mps/load.py]: Feel free to add some type annotations here (i.e `fname` should be `Union[str, Path]`, `ext` should be `str` and `channel` should be `Optional[int]`).

File: tests/test_czi_channel.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import pytest
+
+from mps.load import _format_czi_frames
+
+
+def test_format_czi_frames_selects_channel_from_4d_data():
+    num_frames = 5
```
Comment (finsberg) [inline tests/test_czi_channel.py:137]: This is a good example because your implementation for `_format_czi_frames` assumes that the channel has length between 0 and 10. A potential corner case here would be when one e.g have `(y, channel, x, time)` with `num_frames = num_channels`. Then the second axis will be treated as the time axis. I think the chances for this to happen is quite small, but perhaps one should do a check for this? At least a tests should be implemented that tests this corner case.

File: tests/test_czi_channel.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import numpy as np
+import pytest
+
+from mps.load import _format_czi_frames
+
+
```
Comment (finsberg) [inline tests/test_czi_channel.py:135]: Also add a test for the standard 3D data, and it would also be good to add a test for the full `load_czi` function. Here you could e.g use a [Mock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) for `czifile.CziFile`

**Review by finsberg — CHANGES_REQUESTED** [review]

**Comment by hrehmaan** [comment]
Thanks @finsberg  I have addressed the review comments, added the extra CZI tests, fixed the duplicated load_czi logic, and updated the type annotations/path handling.

I have also run pytest and pre-commit locally. I requested a re-review now.

File: src/mps/load.py [hunk 5]
```diff
@@ -484,9 +606,10 @@ def load_tiff(fname):
     )
 
 
-def load_file(fname, ext):
+def load_file(fname: Union[str, Path], ext: str, channel: Optional[int] = None) -> MPSData:
+    fname = Path(fname)
     if ext == ".czi":
-        data = load_czi(fname)
+        data = load_czi(str(fname), channel=channel)
```
Comment (finsberg) [inline src/mps/load.py:612]: Why do we need to convert the filename to a string?

File: src/mps/load.py [hunk 6]
```diff
@@ -258,11 +373,18 @@ def load_czi(fname):
...
+        num_frames=len(time_stamps),
+        channel=channel,
+    )
+
+    info["size_x"] = frames.shape[0]
+    info["size_y"] = frames.shape[1]
 
     return MPSData(
-        frames=np.swapaxes(images, 0, -1),
+        frames=frames,
```
Comment (finsberg) [inline src/mps/load.py:387]: You are missing the `swapaxes` are you sure this is the same behavior as before?

File: tests/test_czi_channel.py [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+from unittest.mock import Mock, patch
+
+from mps.load import _format_czi_frames, load_czi
+
+
+def test_load_czi_with_mocked_czifile():
+    num_frames = 5
+    y = 20
+    x = 30
```
Comment (finsberg) [inline tests/test_czi_channel.py:12]: Great!

File: src/mps/load.py (hunk 5 above)
Comment (hrehmaan) [reply src/mps/load.py:612]: Because `load_file` now converts `fname` to a `Path` object for type consistency, but the bundled `czifile.CziFile` does not seem to accept `Path` objects directly.

When I tested the real command:

mps2mp4 Control_1.czi --channel 1

passing the `Path` object to `load_czi` caused:

ValueError: The first parameter must be a file name, seekable binary stream, or FileHandle

Converting only the CZI path back to `str` before calling `load_czi` fixes this runtime issue while keeping the `Path` conversion for the other loaders.

File: src/mps/load.py (hunk 6 above)
Comment (hrehmaan) [reply src/mps/load.py:387]: Good point. The `swapaxes` is now handled inside `_format_czi_frames`.

For the standard 3D CZI case `(time, y, x)`, the previous code:

np.swapaxes(images, 0, -1)

produced `(x, y, time)`.

The new helper first moves the detected time axis to the last position and then applies:

frames = np.swapaxes(images, 0, 1)

so the final output is still `(x, y, time)`. I also added a standard 3D test to check this behavior.

**Review by finsberg — APPROVED** [review]
